src/document/DocumentCommandHandlers.js

```diff
@@ -903,6 +919,7 @@ define(function (require, exports, module) {
 
             // First, write document's current text to new file
             newFile = FileSystem.getFileForPath(path);
+            newFile._encoding = doc.file._encoding;
```

Comment:
We can actually write the explicitly selected encoding for path even before we do a 'getFileForPath()'. In that way our existing logic of picking up a user preferred encoding will take care of the rest. With the current code we are actually missing the user preference completely for the new file and in next launch of the new file, encoding detection might result in a corruption where the user might have to select a suitable encoding again.